components/Cards/Flashy.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <section class="card" flex column justify-center align-center>
+    <WithRoot :show="link">
```

Comment:
Don't understand the reason for this component.
Why not having a `v-if` / `v-show` in the parent component or if `link` is mandatory just put `link` property as required and like that the parent component will have to manage this with a simple `v-if`.
This adds some unnecessary complexity

Reply:
It's true that it a bit add complexity. But this is a good way to wrap a child node with parent but by a condition, in this case with an `a tag`. 

`v-if` or `v-show` doesn't solve the problem because they conditionally render the parent and child node together where we what this condition for only parent in this scenario.

This problem also can be solved by separating this component into two and rendering the card with `a tag` or without it by using both `v-if` and `v-else` but again, it requires the component split into two.

And if we add `a tag` on the actual caller component of card component like you said, we can solve it by putting extra few lines but, I like the idea of making `a tag` a configurable thing as a feature of the card component itself.

I don't mind going with your way too but to me, the complexity of `WithRoot` is in an acceptable level.

here is more explanation about the `WithRoot` component: https://forum.vuejs.org/t/conditionally-render-parent-element/9324/2

Reply:
If i understand correctly the code, we don't display the card if there is no link and that's my point why not in that case put the `link` as required in the `props` and like that we have an error if the `link` is empty and the parent has to manage this if needed with a `v-if`.

The component should not be responsible for its rendering. As long as we call it it can be rendered. The parent has control on the child not the other way around.

Also if you really want to go that way for some specific reason (eg: conditional rendering based on `oneOf` properties for example) I really recommend to use directive instead of component as it is a behavior and not a component.

Reply:
No, card is displayed without a link too. If no link provided, `a tag (is the parent)` won't be rendered but its content will.

I don't mind using a directive here but, I don't think that it will feel less complicated for this specific case because we're actually going to not render the html tag that directive is defined on.

Reply:
My bad... didn't really understood what it was doing. I was thinking it was hiding the parent (so the section here) but no it's the child node but still display the children of this child node.

what about something like that ? I feel it's a lot easier to understand
```html
<a target="blank" :href="link" :download="linkFileName" only-children-if="!link">
```

Reply:
Oh yes, with this kind of naming convention `only-children-if="!link"` it's really good. I'll do the changes. 👍 

Infact, we can use `show-only-children="!link"` which would feel very close to `v-show` directive.